crates/kitsune_p2p/proxy/src/tx2.rs

```diff
@@ -0,0 +1,598 @@
+#![allow(clippy::new_ret_no_self)]
+#![allow(clippy::blocks_in_if_conditions)]
+//! Next-gen performance kitsune transport proxy
+
+use crate::*;
+use futures::future::BoxFuture;
+use futures::stream::{Stream, StreamExt};
+use kitsune_p2p_types::tx2::tx2_backend::*;
+use kitsune_p2p_types::tx2::tx2_frontend::tx2_frontend_traits::*;
+use kitsune_p2p_types::tx2::tx2_frontend::*;
+use kitsune_p2p_types::tx2::tx2_utils::*;
+use kitsune_p2p_types::tx2::*;
+use kitsune_p2p_types::*;
+use std::collections::HashMap;
+
+/// Wrap a tx2 backend transport with proxy logic.
+pub fn tx2_proxy(sub_fact: EpFactory, tls_config: TlsConfig) -> EpFactory {
+    ProxyEpFactory::new(sub_fact, tls_config)
+}
+
+// -- private -- //
+
+const PROXY_HELLO_DIGEST: u8 = 0x20;
+const PROXY_FWD_MSG: u8 = 0x30;
+
+struct ProxyConHnd {
+    uniq: Uniq,
+    sub_con: ConHnd,
+    local_digest: CertDigest,
+    remote_digest: CertDigest,
+}
+
+impl std::fmt::Debug for ProxyConHnd {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_tuple("ConHnd").finish()
```

Comment:
Might be useful to include the uniq here for debugging?

Reply:
Good idea, done.